sdk/search/azure-search/src/main/java/com/azure/search/SearchServiceClientBuilder.java

```diff
@@ -0,0 +1,138 @@
+ * {@link SearchServiceClient SearchServiceClients} and {@link SearchServiceAsyncClient SearchServiceAsyncClients}.
+ * Call {@link #buildClient() buildClient} and {@link #buildAsyncClient() buildAsyncClient} respectively to construct
+ * an instance of the desired client.
+ *
+ * <p>
+ * The following information must be provided on this builder:
+ *     <ul>
+ *         <li>the Azure Cognitive Search service endpoint through {@code .endpoint()}
+ *         <li>the API key through {@code .credential()}</li>
+ *     </ul>
+ * </p>
+ */
+@ServiceClientBuilder(serviceClients = {SearchServiceClient.class, SearchServiceAsyncClient.class})
+public class SearchServiceClientBuilder extends SearchClientBuilder {
+    private final ClientLogger logger = new ClientLogger(SearchServiceClientBuilder.class);
+
+    /**
+     * Default Constructor
+     */
+    public SearchServiceClientBuilder() {
+        init();
+    }
+
+    /**
+     * Sets the api version to work against
+     *
+     * @param apiVersion api version
+     * @return the updated SearchServiceClientBuilder object
+     */
+    public SearchServiceClientBuilder apiVersion(SearchServiceVersion apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    /**
+     * Sets the Azure Cognitive Search service endpoint
+     *
+     * @param endpoint the endpoint URL to the Azure Cognitive Search service
+     * @return the updated SearchIndexClientBuilder object
+     */
+    public SearchServiceClientBuilder endpoint(String endpoint) {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Set the http client (optional). If this is not set, a default httpClient will be created
+     *
+     * @param httpClient value of httpClient
+     * @return the updated SearchServiceClientBuilder object
+     */
+    public SearchServiceClientBuilder httpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Sets the api key to use for request authentication.
+     *
+     * @param searchApiKeyCredential api key for request authentication
+     * @return the updated SearchServiceClientBuilder object
+     * @throws IllegalArgumentException when the api key is empty
+     */
+    public SearchServiceClientBuilder credential(SearchApiKeyCredential searchApiKeyCredential) {
+        if (searchApiKeyCredential == null || StringUtils.isBlank(searchApiKeyCredential.getApiKey())) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Empty apiKeyCredentials"));
+        }
+        this.searchApiKeyCredential = searchApiKeyCredential;
+        return this;
+    }
+
+    /**
+     * Http Pipeline policy
+     *
+     * @param policy policy to add to the pipeline
+     * @return the updated SearchServiceClientBuilder object
+     */
+    public SearchServiceClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        this.policies.add(policy);
```

Comment:
Generally `addPolicy` verifies that the `HttpPipelinePolicy` isn't `null` and if it is it throws an exception.